TestStack.Dossier.Tests/BuilderBuildListTests.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            var builders = Builder<Customer>.CreateListOfSize(5);
+
+            var entities = builders.BuildList();
+
+            entities.Count.ShouldBe(5);
+        }
+
+        [Fact]
+        public void GivenListOfBuilders_WhenCallingBuildListImplicitly_ThenAListOfEntitiesOfTheRightSizeShouldBeReturned()
+        {
+            List<Customer> entities = Builder<Customer>.CreateListOfSize(5);
+
+            entities.Count.ShouldBe(5);
+        }
+
+        [Fact]
+        public void GivenListOfBuilders_WhenCallingBuildListExplicitly_ThenAListOfEntitiesOfTheRightTypeShouldBeReturned()
+        {
+            var builders = Builder<Customer>.CreateListOfSize(5);
+
+            var entities = builders.BuildList();
+
+            entities.ShouldBeAssignableTo<IList<Customer>>();
+        }
+
+        [Fact]
+        public void GivenListOfBuilders_WhenCallingBuildListImplicitly_ThenAListOfEntitiesOfTheRightTypeShouldBeReturned()
+        {
+            List<Customer> entities = Builder<Customer>.CreateListOfSize(5);
+
+            entities.ShouldBeAssignableTo<IList<Customer>>();
+        }
+
+        [Fact]
+        public void GivenListOfBuilders_WhenCallingBuildListExplicitly_ThenAListOfUniqueEntitiesShouldBeReturned()
+        {
+            var builders = Builder<Customer>.CreateListOfSize(5);
+
+            var entities = builders.BuildList();
+
+            entities[0].ShouldNotBe(entities[1]);
+            entities[0].ShouldNotBe(entities[2]);
+            entities[0].ShouldNotBe(entities[3]);
+            entities[0].ShouldNotBe(entities[4]);
+            entities[1].ShouldNotBe(entities[2]);
+            entities[1].ShouldNotBe(entities[3]);
+            entities[1].ShouldNotBe(entities[4]);
+            entities[2].ShouldNotBe(entities[3]);
+            entities[2].ShouldNotBe(entities[4]);
+            entities[3].ShouldNotBe(entities[4]);
+        }
+
+        [Fact]
+        public void GivenListOfBuilders_WhenCallingBuildListImplicitly_ThenAListOfUniqueEntitiesShouldBeReturned()
+        {
+            List<Customer> entities = Builder<Customer>.CreateListOfSize(5);
+
+            entities[0].ShouldNotBe(entities[1]);
+            entities[0].ShouldNotBe(entities[2]);
+            entities[0].ShouldNotBe(entities[3]);
+            entities[0].ShouldNotBe(entities[4]);
+            entities[1].ShouldNotBe(entities[2]);
+            entities[1].ShouldNotBe(entities[3]);
+            entities[1].ShouldNotBe(entities[4]);
+            entities[2].ShouldNotBe(entities[3]);
+            entities[2].ShouldNotBe(entities[4]);
+            entities[3].ShouldNotBe(entities[4]);
+        }
+
+        [Fact]
+        public void GivenListOfBuildersWithCustomisation_WhenBuildingEntitiesExplicitly_ThenTheCustomisationShouldTakeEffect()
+        {
+            var generator = new SequentialGenerator(0, 100);
+            var list = CustomerBuilder.CreateListOfSize(3)
```

Comment:
These were a copy of the BuilderListTests, using Builder<Customer> in place of CustomerBuilder. As part of the design process I wanted to ensure that the new Builder would have the same functionality as a CustomerBuilder by having it satisfy the same specification.